storage/aws/efs/variables.tf

```diff
@@ -5,46 +5,84 @@ variable "tags" {
   default     = {}
 }
 
-# VPC info
-variable "vpc" {
-  description = "AWS VPC info"
-  type = object({
-    id                 = string
-    cidr_blocks        = set(string)
-    cidr_block_private = set(string)
-    subnet_ids         = set(string)
-  })
+# VPC infos
+variable "vpc_id" {
+  description = "AWS VPC id"
+  type        = string
+}
+
+variable "vpc_cidr_blocks" {
+  description = "AWS VPC cidr block"
+  type        = set(string)
+}
+
+variable "vpc_cidr_block_private" {
+  description = "AWS VPC private cidr block "
+  type        = set(string)
+}
+
+variable "vpc_subnet_ids" {
+  description = "AWS VPC subnet ids"
+  type        = set(string)
```

Comment:
What is the reason behind this change? I think it is much simpler to deal with objects as long as the objects are constructed from other modules.

I would much prefer have this:
```hcl
module "my_vpc" {
  ...
}
module "my_efs" {
  source = ...
  vpc = module.my_vpc
  ...
}
```
rather than this:
```hcl
module "my_vpc" {
  ...
}
module "my_efs" {
  source = ...
  vpc_id = module.my_vpc.id
  vpc_cidr_blocks = module.my_vpc.cidr_blocks
  vpc_cidr_block_private = module.my_vpc.cidr_block_private
  vpc_subnet_ids = module.my_vpc.subnet_ids
  ...
}
```

Of course, this implies to have a well defined module schema.
Although, I have no problem removing the nested structure `efs` as it is the role of this very module to create the EFS.